src/pytorch_lightning/core/module.py

```diff
@@ -266,7 +273,7 @@ def logger(self) -> Optional[Logger]:
         # this should match the implementation of `trainer.logger`
         # we don't reuse it so we can properly set the deprecation stacklevel
         if self._trainer is None:
-            return
+            return  # type: ignore[return-value]
```

Comment:
see https://github.com/Lightning-AI/lightning/pull/13603#discussion_r929958803

Reply:
I don't understand what the problem is here. There is nothing to ignore here. Mypy just wants you to write explicitly None, instead of an implicit blank return.